deps/github.com/arangodb/arangosync/client/api.go

```diff
@@ -0,0 +1,333 @@
+	Outgoing       []OutgoingSyncInfo `json:"outgoing,omitempty"`       // Status of outgoing synchronization
+	MessageTimeout time.Duration      `json:"messageTimeout,omitempty"` // Maximum time between messages in a task channel
+}
+
+// OutgoingSyncInfo holds JSON info returned as part of `GET /_api/sync`
+// regarding a specific target for outgoing synchronization data.
+type OutgoingSyncInfo struct {
+	ID       string          `json:"id"`               // ID of sync master to which data is being send
+	Endpoint Endpoint        `json:"endpoint"`         // Endpoint of sync masters to which data is being send
+	Status   SyncStatus      `json:"status"`           // Overall status for this outgoing target
+	Shards   []ShardSyncInfo `json:"shards,omitempty"` // Status of outgoing synchronization per shard for this target
+}
+
+// ShardSyncInfo holds JSON info returned as part of `GET /_api/sync`
+// regarding a specific shard.
+type ShardSyncInfo struct {
+	Database              string        `json:"database"`                 // Database containing the collection - shard
+	Collection            string        `json:"collection"`               // Collection containing the shard
+	ShardIndex            int           `json:"shardIndex"`               // Index of the shard (0..)
+	Status                SyncStatus    `json:"status"`                   // Status of this shard
+	StatusMessage         string        `json:"status_message,omitempty"` // Human readable message about the status of this shard
+	Delay                 time.Duration `json:"delay,omitempty"`          // Delay between other datacenter and us.
+	LastMessage           time.Time     `json:"last_message"`             // Time of last message received by the task handling this shard
+	LastDataChange        time.Time     `json:"last_data_change"`         // Time of last message that resulted in a data change, received by the task handling this shard
+	LastShardMasterChange time.Time     `json:"last_shard_master_change"` // Time of when we last had a change in the status of the shard master
+	ShardMasterKnown      bool          `json:"shard_master_known"`       // Is the shard master known?
+}
+
+type SyncStatus string
+
+const (
+	// SyncStatusInactive indicates that no synchronization is taking place
+	SyncStatusInactive SyncStatus = "inactive"
+	// SyncStatusInitializing indicates that synchronization tasks are being setup
+	SyncStatusInitializing SyncStatus = "initializing"
+	// SyncStatusInitialSync indicates that initial synchronization of collections is ongoing
+	SyncStatusInitialSync SyncStatus = "initial-sync"
+	// SyncStatusRunning indicates that all collections have been initially synchronized
+	// and normal transaction synchronization is active.
+	SyncStatusRunning SyncStatus = "running"
+	// SyncStatusCancelling indicates that the synchronization process is being cancelled.
+	SyncStatusCancelling SyncStatus = "cancelling"
```

Comment:
preferred spelling is one "l": canceling ... I was told by CTO at NuView to "fix it everywhere".  Not sure how English spelling rules apply at ArangoDB.